File: cgi-bin/cgi-src/telephone.cpp
#include <stdio.h>
#include <string>

int main(int argc, char** argv, char** env)
{
	std::string	name;
	std::string	var;

	for (int i = 0; env[i] != NULL; i++) {
		var = env[i];
		if (var.rfind("QUERY_STRING=", 0) != std::string::npos)
		{
			var.erase(0, var.find('=')+1);
			while (!var.empty())
			{
				if (var.rfind("name=", 0) != std::string::npos) {
					var.erase(0, 5);
					name = var.substr(0, var.find('&'));
					break;
				}
				if (var.find('&') == std::string::npos)
					break;
				var.erase(0, var.find('&')+1);
			}
			break ;
		}
	}

	if (name.empty())
		name = "<Unknown>";

	printf("Content-type: text/plain\n\n");
	printf("\n");
	printf("   //\\\n");
	printf("  | \\/ ~~ Hello %s, CPP-written CGI speaking?\n", name.c_str());
	printf("  ||~                                             _     _\n");
	printf("  ||_                                            [ L___I ]\n");
	printf("  | /\\                                          |   ...   |\n");
	printf(" ,@\\\\/    ,@@@,            ,@@@@@,              |   :::   |\n");
	printf(" @,    ,@@\"   \"@@@,     ,@@\"     \"@@@,    ,@@@@\"|   '''   |\n");
	printf(" \"@@@@@\"          \"@@@@@\"            \"@@@@\"     '========='\n");
	printf("\n");
}

File: cgi-bin/telephone.py
import os
import sys

def print_phone(the_name):
	print("Content-type: text/plain\n\n")
	print("")
	print("   //\\")
	print("  | \\/ ~~ Hello " + the_name + ", Python CGI speaking?")
	print("  ||~                                             _     _")
	print("  ||_                                            [ L___I ]")
	print("  | /\\                                          |   ...   |")
	print(" ,@\\\\/    ,@@@,            ,@@@@@,              |   :::   |")
	print(" @,    ,@@\"   \"@@@,     ,@@\"     \"@@@,    ,@@@@\"|   '''   |")
	print(" \"@@@@@\"          \"@@@@@\"            \"@@@@\"     '========='")
	print("")

QUERY_STRING = os.getenv('QUERY_STRING')

if QUERY_STRING is None:
	print_phone("<Unknown>")
	sys.exit(0)

if QUERY_STRING.find('name') == -1:
	sys.exit(1)

array = QUERY_STRING.split('=')

final_name = ""

for (index, element) in enumerate(array):
	print(element)
	if element.find('name') != -1:
		name = array[index + 1]
		if name.find('&') != -1:
			final_name = name[:name.index('&')]
		else:
			final_name = name
		break

print_phone(final_name)